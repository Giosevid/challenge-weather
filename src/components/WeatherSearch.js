import React from 'react'
import { Formik, Form, Field } from 'formik';
import { Button } from "react-bootstrap";
import SignupSchema from "../config/validationsForm";
import { getWeatherActions, cleanWeather } from "../redux/weatherDuck";
import { connect } from "react-redux"

const WeatherSearch = ({ getWeatherActions, cleanWeather, cityName }) => {

  const initialValues = {
    cityName: cityName || ''
  }

  const cleanInput = setFieldValue => {
    setFieldValue('cityName', '')
    cleanWeather()
  }

  const onSubmit = ({ cityName, setErrors }) => getWeatherActions(cityName, setErrors)

  return(
    <article className="jumbotron text-center">
      <section className="container">
        <h1 className="jumbotron-heading">¡Busca tu ciudad!</h1>
        <p className="lead text-muted">Conoce toda la información del clima referente a tu ciudad en un solo lugar.</p>
        <Formik initialValues={initialValues} onSubmit={onSubmit} validationSchema={SignupSchema}>
          {({ errors, touched, setFieldValue }) => (
            <Form>
              <Field name="cityName" type="text" className="form-control text-center" placeholder="Ingresa tu ciudad" />
                {errors.search && touched.search && <div>{errors.search}</div>}
              <div className="wrap-button">
                <Button type="submit" variant="primary">Consultar</Button>{' '}
                <Button variant="light" onClick={() => cleanInput(setFieldValue)}>Limpiar</Button>
              </div>
            </Form>
          )}
        </Formik>
      </section>
    </article>
  )
}

export const mapStateToProps = ({weather}) => ({
  cityName: weather?.currentWeather?.name
})

export default connect(mapStateToProps, {getWeatherActions,cleanWeather})(WeatherSearch)
