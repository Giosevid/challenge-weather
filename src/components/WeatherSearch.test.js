import React from "react";
import { shallow } from "enzyme";
import WeatherSearch, { mapStateToProps } from "./WeatherSearch";
import Enzyme from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import configureStore from "redux-mock-store";

Enzyme.configure({ adapter: new Adapter() });

const middlewares = [];
let mockStore,
  initialState = { currentWeather: "" };

const setup = () => {
  return shallow(<WeatherSearch store={mockStore(initialState)} />);
};

describe("testing compoenent WeatherSearch", () => {
  beforeEach(() => {
    mockStore = configureStore(middlewares);
  });
  test("renders without crashing", () => {
    const wrapper = setup();
    expect(wrapper.length).toBe(1);
    expect(wrapper).toMatchSnapshot();
  });
});
