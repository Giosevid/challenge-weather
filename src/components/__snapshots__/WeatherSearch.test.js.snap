// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing compoenent WeatherSearch renders without crashing 1`] = `ShallowWrapper {}`;
